### ISSUE TITLE
`unicorn/better-regex` conflicts with `regexp/strict` and `regexp/match-any`

### ISSUE THREAD

**Issue by JounQin** [sindresorhus/eslint-plugin-unicorn#1852]
### Description

**Information:**
- ESLint version: v8.19.0
- `eslint-plugin-regexp` version: 1.7.0

**Description**

```js
/\[([^[\]]+)]/g
```

error from this plugin

```log
Unescaped source character ']'
```

After fixed:

```js
/\[([^[\]]+)\]/g
```

error from `unicorn/better-regex`

```log
/\[([^[\]]+)\]/g can be optimized to /\[([^[\]]+)]/g
```

```js
/{{([\S\s]+?)}}/g
```

`regexp/match-any` is reported

```js
/{{([\s\S]+?)}}/g
```

`unicorn/better-regex` is reported

### Fail

As above

### Pass

As above

### Additional Info

I hope these two plugins can work better together by default, some rule can be extracted from `unicorn` to `regexp` instead?

**Comment by JounQin** [comment]
A related previous issue #469, we need to clarify why current rule `unicon/better-regex` is `better`.

**Comment by sindresorhus** [comment]
It was removed: https://github.com/sindresorhus/eslint-plugin-unicorn/commit/89dc5dc04389929ae786e171bac2af522fe2e7d8
